### PR TITLE
Created individual project page

### PR DESCRIPTION
**Pull request by anu-shaK**
1. Created an api route to fetch a project with its project ID.
2. Display project description, funding details and user name.

### CODE REVIEW

File: client/src/components/IndividualProject.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            <Box className={classes.fundingDetails}>
+              <Grid item xs={5} className={classes.stats}>
+                <Typography component="h1" variant="h4">
+                  {investors.length}
+                </Typography>
+                <Typography variant="subtitle1">backers</Typography>
+              </Grid>
+              {/* <Grid item xs={2}>
+                <Divider orientation="vertical" />
+              </Grid> */}
```
Comment (skassam21) [inline client/src/components/IndividualProject.js]: Remove this unused commented out code

File: server/routes/projectRoutes.js [hunk 2]
```diff
@@ -9,7 +10,8 @@ const {
 const auth = require('../middleware/authMiddleware');
 
 router.get('/', getAllProjects); // Get all projects (public)
-router.get('/:userId/', getProjectsForUser); // Get all projects for a specific user
+router.get('/:projectId', getProject);
+//router.get('/:userId', getProjectsForUser); // Get all projects for a specific user
```
Comment (skassam21) [inline server/routes/projectRoutes.js:14]: Is there a reason you removed this route?

File: server/controllers/projectController.js [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
+const Project = require('../models/projects');
+const User = require('../models/User');
+
+async function getProject(req, res) {
+  const project = await Project.findOne({ _id: req.params.projectId });
+  if (!project) {
+    return res.status(400);
+  }
+  const user = await User.findOne({ _id: project.owner }).select({ name: 1 });
+  res.send({ project, user });
```
Comment (skassam21) [inline server/controllers/projectController.js]: I think you can use the mongoDB populate to get the owner information inside of the project data

File: client/src/pages/ProjectDetails.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    getProjectDetails();
+  }, []);
+
+  if (errors) {
+    return <h1>Error retrieving project</h1>;
+  }
+
+  return (
+    <>
+      {project && user && <IndividualProject project={project} user={user} />}
```
Comment (skassam21) [inline client/src/pages/ProjectDetails.js]: We should eventually add loading states

File: client/src/pages/ProjectDetails.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from 'react';
+import IndividualProject from '../components/IndividualProject';
+
+const projectID = window.location.pathname.split('/').pop();
```
Comment (skassam21) [inline client/src/pages/ProjectDetails.js]: I'd prefer you'd get the projectID from using React Router, so this properly updates when the component updates 

You can do that using a React Router hook

File: server/controllers/projectController.js (hunk 3 above)
Comment (anu-shaK) [reply server/controllers/projectController.js]: Okay, modified to populate and changed on the front end.

File: client/src/pages/ProjectDetails.js (hunk 5 above)
Comment (anu-shaK) [reply client/src/pages/ProjectDetails.js]: Okay, changed to useParams() to get project id.

File: server/routes/projectRoutes.js (hunk 2 above)
Comment (anu-shaK) [reply server/routes/projectRoutes.js:14]: Actually it was creating a conflict between getProject and getProjectForUser, both being get requests. Maybe we can change id to a query parameter?

File: client/src/components/IndividualProject.js (hunk 1 above)
Comment (anu-shaK) [reply client/src/components/IndividualProject.js]: My bad, deleted it!

File: client/src/pages/ProjectDetails.js (hunk 4 above)
Comment (anu-shaK) [reply client/src/pages/ProjectDetails.js]: Okay, added loading states.

**Review by skassam21 — APPROVED** [review]

**Pull request merged**